loopy/check.py

```diff
@@ -1207,6 +1207,124 @@ def check_that_shapes_and_strides_are_arguments(kernel):
 # }}}
 
 
+# {{{ validate_kernel_call_sites
+
+def _are_sub_array_refs_equal(sar1, sar2, caller):
```

Comment:
Why/how is this different from `SubArrayRef.__eq__`?

Reply:
Explained in 4d47b445c134f53dd8392899703fec0f00f0748d.